src/operator/cudnn_rnn-inl.h

```diff
@@ -446,20 +456,19 @@ class CuDNNRNNOp : public Operator{
                                             strideA));
 
       // Create Dropout descriptors
-      CUDNN_CALL(cudnnCreateDropoutDescriptor(&dropout_desc_));
       CUDNN_CALL(cudnnDropoutGetStatesSize(s->dnn_handle_,
                                            &dropout_byte_));
-      dropout_size_ = dropout_byte_ / sizeof(DType);
-      dropout_states_ = Storage::Get()->Alloc(dropout_byte_, Context::GPU());
+      if (param_.p > 0) {
```

Comment:
is it necessary to get dropout state bytes if `param_.p` is 0?

Reply:
According to https://docs.nvidia.com/deeplearning/sdk/cudnn-developer-guide/index.html#cudnnSetDropoutDescriptor `states` can be null. It is not documented what happens then and when this should be the case, but presumably only for `p == 0` it is sensible. Ie for `p == 0` there is no need for randomness.